DataGrid/Extension/Symfony/EventSubscriber/FormSubscriber.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+/**
+ * (c) FSi sp. z o.o. <info@fsi.pl>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace FSi\Bundle\DataGridBundle\DataGrid\Extension\Symfony\EventSubscriber;
+
+use FSi\Bundle\DataGridBundle\DataGrid\Extension\Symfony\ColumnTypeExtension\CellFormBuilder;
+use FSi\Bundle\DataGridBundle\Form\Type\RowType;
+use FSi\Component\DataGrid\Column\CellViewInterface;
+use FSi\Component\DataGrid\DataGrid;
+use FSi\Component\DataGrid\DataGridEvent;
+use FSi\Component\DataGrid\DataGridEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class FormSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var FormFactoryInterface
+     */
+    private $formFactory;
+
+    /**
+     * @var bool
+     */
+    private $csrfTokenEnabled;
+
+    /**
+     * @var FormInterface[]
+     */
+    private $forms;
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            DataGridEvents::POST_SET_DATA => 'buildForms',
+        ];
+    }
+
+    public function __construct(FormFactoryInterface $formFactory, bool $csrfTokenEnabled = true)
+    {
+        $this->formFactory = $formFactory;
+        $this->csrfTokenEnabled = $csrfTokenEnabled;
+        $this->forms = [];
+    }
+
+    public function buildForms(DataGridEvent $event): void
+    {
+        $dataGrid = $event->getDataGrid();
+        $data = $event->getData();
+
+        $entryType = $this->isSymfony3() ? RowType::class : 'row';
+
+        if ($this->csrfTokenEnabled) {
+            $formBuilderOptions['csrf_protection'] = false;
+        }
+
+        $formBuilder = $this->formFactory
+            ->createNamedBuilder($dataGrid->getName(), $this->getFormTypeName(), $data, $formBuilderOptions);
+
+        foreach ($data as $index => $row) {
+            $formBuilder->add($index, $entryType, [
+                'data_class' => get_class($row),
+                'property_path' => sprintf('[%s]', $index),
+            ]);
+            $rowFormBuilder = $formBuilder->get($index);
+
+            foreach ($dataGrid->getColumns() as $column) {
+                if (!$column->hasOption('editable') || !$column->getOption('editable')) {
```

Comment:
Perhaps we should filter the columns in DataGrid to check if any is actually editable and only then create the form?